src/callbacks_step/averaging_dg2d.jl

```diff
@@ -0,0 +1,84 @@
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+
+# Create arrays with DGSEM-specific structure to store the mean values and set them all to 0
+function initialize_mean_values(mesh::TreeMesh{2}, equations::AbstractCompressibleEulerEquations{2},
+                                dg::DGSEM, cache)
+  uEltype = eltype(cache.elements)
+  v_mean = zeros(uEltype, (ndims(equations), nnodes(dg), nnodes(dg), nelements(cache.elements)))
+  c_mean = zeros(uEltype, (nnodes(dg), nnodes(dg), nelements(cache.elements)))
+  rho_mean = zeros(uEltype, size(c_mean))
+  vorticity_mean = zeros(uEltype, size(c_mean))
+
+  return (; v_mean, c_mean, rho_mean, vorticity_mean)
+end
+
+# Create cache which holds the vorticity for the previous time step. This is needed due to the
+# trapezoidal rule
+function create_cache(::Type{AveragingCallback}, mesh::TreeMesh{2},
+                      equations::AbstractCompressibleEulerEquations{2}, dg::DGSEM, cache)
+  # Cache vorticity from previous time step
+  uEltype = eltype(cache.elements)
+  vorticity_prev = zeros(uEltype, (nnodes(dg), nnodes(dg), nelements(cache.elements)))
+  return (; vorticity_prev)
+end
+
+# Calculate vorticity for the initial solution and store it in the cache
+function initialize_cache!(averaging_callback_cache, u,
+                           mesh::TreeMesh{2}, equations::AbstractCompressibleEulerEquations{2},
+                           dg::DGSEM, cache)
+  @unpack vorticity_prev = averaging_callback_cache
+
+  # Calculate vorticity for initial solution
+  calc_vorticity!(vorticity_prev, u, equations, dg, cache)
+
+  return nothing
+end
+
+
+# Update mean values using the trapezoidal rule
+function calc_mean_values!(mean_values, averaging_callback_cache, u, u_prev, integration_constant,
+                           mesh::TreeMesh{2}, equations::AbstractCompressibleEulerEquations{2},
+                           dg::DGSEM, cache)
+  @unpack v_mean, c_mean, rho_mean, vorticity_mean = mean_values
+  @unpack vorticity_prev = averaging_callback_cache
+
+  @threaded for element in eachelement(dg, cache)
+    for j in eachnode(dg), i in eachnode(dg)
+      vorticity = calc_vorticity_node(u, equations, dg, cache, i, j, element)
+      vorticity_prev_node = vorticity_prev[i, j, element]
+      vorticity_prev[i, j, element] = vorticity # Cache current velocity for the next time step
+
+      u_node_prim = cons2prim(get_node_vars(u, equations, dg, i, j, element), equations)
+      u_prev_node_prim = cons2prim(get_node_vars(u_prev, equations, dg, i, j, element), equations)
+
+      rho = u_node_prim[1]
+      v1 = u_node_prim[2]
+      v2 = u_node_prim[3]
+      p = u_node_prim[4]
+
+      rho_prev = u_prev_node_prim[1]
+      v1_prev = u_prev_node_prim[2]
+      v2_prev = u_prev_node_prim[3]
+      p_prev = u_prev_node_prim[4]
```

Comment:
```suggestion
      rho,      v1,      v2,      p      = u_node_prim
      rho_prev, v1_prev, v2_prev, p_prev = u_prev_node_prim
```
That's shorter and requires less attention when reading it (to check whether the indices `[1]`, `[2]`, `[3]`, `[4]` are in order).